src/coreclr/src/jit/gentree.cpp

```diff
@@ -6076,7 +6076,7 @@ GenTree* Compiler::gtNewStringLiteralNode(InfoAccessType iat, void* pValue)
                 // Non CoreRT - This case is illegal, creating a TYP_REF from an INT_CNS
                 noway_assert(!"unreachable IAT_VALUE case in gtNewStringLiteralNode");
```

Comment:
I think this no-way assert can be deleted.

Reply:
Why do you prefer it to be deleted?
I would prefer to keep it as is or replace it with a plain assert, more like a reminder that only CoreRT supports it and CoreCLR can be improved.

Reply:
It violates the JIT/EE interface abstraction. Abstraction violations like this are what makes the system hard to change.

We had similar discussion in https://github.com/dotnet/runtime/pull/42021#discussion_r491412446 .

Reply:
Got it, deleted. Thank you for the explanation.